### ISSUE TITLE
`no-setup-props-reactivity-loss` should report template string expressions

### ISSUE THREAD

**Issue by ByScripts** [vuejs/eslint-plugin-vue#2470]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.57.0
- **eslint-plugin-vue version:** 9.26.0
- **Vue version:** 3.4.27
- **Node version:** 20.12.2
- **Operating System:** macOS Sonoma 14.5

**Please show your full configuration:**

```json5
{
  rules: {
    'vue/no-setup-props-reactivity-loss': 'error'
  }
}
```

**What did you do?**	

```vue
<template>{{ foo }}</template>

<script setup lang="ts">
const props = defineProps<{
  foo: string
}>()

const foo = `${props.foo}`
</script>
```

**What did you expect to happen?**

Following code should throw an ESLint error

```ts
const foo = `${props.foo}`
```

**What actually happened?**

No error

**Link to reproduce this issue**

https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxdz9sKwjAMBuBXCUXYBOvAS083IvgQu9jYMizMpKTtUErf3VJExLscvsCfqAYeUe3V0Q1irD+3ZB6WxUOEESdDeOHcE5KHBJPwA6olYHVoCZ/FZdWH2f/rOrYE4NAHW1th69ZQJgADk/Nw7xe8irDACcp+OzEffoW7c5jH24/rVvFLU1dw2rSU1rk8Np8H1EZJmNGpfVQ5aUOsSwpdTrVgP3izGP/SM7usdim9AQOUXZw=

**Comment by FloEdelmann** [comment]
Pull request welcome to fix this false-negative!
